src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -396,8 +396,10 @@ public async Task ReadResultToEnd(DbDataReader dbDataReader, CancellationToken c
                     //
                     availableTask = SendCurrentResults();
 
-                    while (await dataReader.ReadAsync(cancellationToken))
+                    while (dataReader.Read())
```

Comment:
Please add a comment explaining why you're using read vs readasync. is there an issue/doc you can link to about the perf issue? That is not intuitive to me at all (generally using async calls is preferred for async code like STS) so it's important to make sure future people know why a particular choice was made.